spec/unit/module_loader/puppetfile_spec.rb

```diff
@@ -276,6 +276,30 @@ def expect_wrapped_error(error, pf_path, error_type)
       end
     end
 
+    describe 'forge declaration' do
+      before(:each) do
+        PuppetForge.host = ""
+      end
+
+      it 'is respected if `allow_puppetfile_override` is true' do
+        @path = File.join(PROJECT_ROOT, 'spec', 'fixtures', 'unit', 'puppetfile', 'forge-override')
+        puppetfile = R10K::ModuleLoader::Puppetfile.new(basedir: @path, overrides: { forge: { allow_puppetfile_override: true } })
+        puppetfile.load!
+        expect(puppetfile.instance_variable_get(:@forge)).to eq("my.custom.forge.com")
+        expect(PuppetForge.host).to eq("my.custom.forge.com/")
+      end
+
+      it 'is ignored if `allow_puppetfile_override` is false' do
+        @path = File.join(PROJECT_ROOT, 'spec', 'fixtures', 'unit', 'puppetfile', 'forge-override')
+        puppetfile = R10K::ModuleLoader::Puppetfile.new(basedir: @path, overrides: { forge: { allow_puppetfile_override: false } })
+        expect(PuppetForge).not_to receive(:host=).with("my.custom.forge.com")
+        puppetfile.load!
+        # instance variable is set, but PuppetForge.host is not updated
+        expect(puppetfile.instance_variable_get(:@forge)).to eq("my.custom.forge.com")
+        expect(PuppetForge.host).to eq("/")
```

Comment:
Why doesn't this equal the default Forge value? It seems super weird to me to have various places where we set `PuppetForge.host` and that those places are different from where that host is actually used??

Reply:
This is an artifact of the test: this is a unit test of the `ModuleLoader::Puppetfile`, so the thing that would normally set it (settings init) doesn't happen here. If you notice above, I've provided a default value in a `before` block. If I didn't do that, the state bled between tests, because `host` is a class variable of `PuppetForge`. I could use a _different_ default value, but that doesn't change the nature of the test.

Reply:
oh right, I forgot you were resetting it by the time I got to this part haha. so if it was a single tests where you didn't have the `before` block, `PuppetForge.host` would be the default value? Is that our default or the forge gem default?

Reply:
It would be the Forge gem's default, because even with one test, our code that sets it is not getting executed in this context. I don't know if that default is something valid, or just `nil`, I'd have to check.